### PR TITLE
feat(feed-item-row): recipe for feed item row - vue3

### PR DESCRIPTION
**Pull request by sanjeevdialpad**
# Feed Item Row

## :hammer_and_wrench: Type Of Change

- [ ] Fix
- [x] Feature
- [ ] Refactoring
- [x] Documentation

## :book: Description

Feed Item row recipe which can be used for any row in a feed list. This comes with slots for threads, menu, reactions and default content slots. It uses list item component, Dt Avatar, Dt Lazy show to render different sections of the feed item. Further reactions, thread component created in future will fit into the respective slots.

## :pencil: Checklist

- [x] I have reviewed my changes
- [ ] I have added tests
- [x] I have added all relevant documentation
- [x] I have validated components with a screen reader
- [x] I have validated components keyboard navigation
- [x] I have considered the performance impact of my change
- [x] I have checked that my change did not significantly increase bundle size
- [x] I am exporting any new components or constants in the index.js in the component directory
- [x] I am exporting any new components or constants in the index.js in the root

## :crystal_ball: Next Steps

> Tests
> Integrate feed item row in DP 2.0.
> Create the thread and reactions component and replace the respective slots with default content in this recipe.

## :camera: Screenshots / GIFs

Screenshot 2023-04-20 at 11 44 53 AM
Screenshot 2023-04-20 at 11 45 15 AM

## :link: Sources

https://www.figma.com/file/oTY56LgQha8ljb4c8w4GtO/DT-Shared%3A-App-Components?node-id=9240-7140&t=yloR0A6sUc0Ko77p-0

### CODE REVIEW

File: recipes/conversation_view/feed_item_row/feed_item_row_default.story.vue [hunk 1]
```diff
@@ -0,0 +1,97 @@
+<template>
+  <ul>
+    <dt-recipe-feed-item-row
+      :show-header="showHeader"
```
Comment (braddialpad) [inline recipes/conversation_view/feed_item_row/feed_item_row_default.story.vue]: All of these types of references in .story.vue files need to become like:

```
:show-header="$attrs.showHeader"
```

File: recipes/conversation_view/feed_item_row/feed_item_row.stories.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+const DefaultTemplate = (args, { argTypes }) => createTemplateFromVueFile(
+  args,
+  argTypes,
+  DtRecipeFeedItemRowDefaultTemplate,
+);
+const VariantsTemplate = (args, { argTypes }) => createTemplateFromVueFile(
+  args,
+  argTypes,
+  DtRecipeFeedItemRowVariantsTemplate,
+);
```
Comment (braddialpad) [inline recipes/conversation_view/feed_item_row/feed_item_row.stories.js]: Remove argTypes so this just becomes like:

```
const DefaultTemplate = (args) => createTemplateFromVueFile(
  args,
  DtRecipeFeedItemRowDefaultTemplate,
);
```

**Review by braddialpad — COMMENTED** [review]
This is not currently rendering in vue 3.

There are a few changes you need to make when you move something over to vue 3, pointed them out below.

**Comment by sanjeevdialpad** [comment]
thanks Brad for all the help! I will open a PR with tests. Waiting for design to give a final go ahead.

**Comment by braddialpad** [comment]
Looks like there's a couple a11y errors

**Comment by sanjeevdialpad** [comment]
@bhawkins-dialpad tagging for design approval

**Review by braddialpad — APPROVED** [review]

**Pull request merged**